### PR TITLE
[raw vehicle cmd converter] apply autoware auto msgs

### PR DESCRIPTION
**Pull request by taikitanaka3**
## Description

apply autoware auto msgs to raw vehicle cmd converter

autoware_iv_msgs
- actuation_command
- steering_report
- float32_multi_array_stamped
- localization odometry

autoware_auto_msgs
- ackermann_control_command

remove unused shift message

## Review Procedure

colcon build pass

## Remarks

 - control command stamped -> ackerman control command
 - vehicle status steering was not in .auto msgs
 - localization/twist would be necessary to change to odm later [not in this PR]

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by taikitanaka3** [comment]
TODO 
use nav_msgs::Odmetry instead of twist

**Comment by taikitanaka3** [comment]
@takayuki5168 
can you review this?

File: vehicle/raw_vehicle_cmd_converter/include/raw_vehicle_cmd_converter/node.hpp [hunk 1]
```diff
@@ -22,24 +22,25 @@
 
 #include <rclcpp/rclcpp.hpp>
 
-#include <autoware_control_msgs/msg/control_command_stamped.hpp>
+#include <autoware_auto_control_msgs/msg/ackermann_control_command.hpp>
 #include <autoware_debug_msgs/msg/float32_multi_array_stamped.hpp>
 #include <autoware_vehicle_msgs/msg/actuation_command_stamped.hpp>
-#include <autoware_vehicle_msgs/msg/shift.hpp>
 #include <autoware_vehicle_msgs/msg/steering.hpp>
```
Comment (tkimura4) [inline vehicle/raw_vehicle_cmd_converter/include/raw_vehicle_cmd_converter/node.hpp:28]: In autoware.auto, VehicleKinematicState seems to be used for getting steering angle.

https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/blob/master/src/control/trajectory_follower_nodes/src/lateral_controller_node.cpp#L159-L162

**Review by tkimura4 — APPROVED** [review]

**Comment by taikitanaka3** [comment]
@TakaHoribe 
merge readyです！

File: vehicle/raw_vehicle_cmd_converter/src/node.cpp [hunk 2]
```diff
@@ -86,8 +86,8 @@ RawVehicleCommandConverterNode::RawVehicleCommandConverterNode(
   pub_actuation_cmd_ = create_publisher<ActuationCommandStamped>("/vehicle/actuation_cmd", 1);
   sub_control_cmd_ = create_subscription<ControlCommandStamped>(
     "/control/control_cmd", 1, std::bind(&RawVehicleCommandConverterNode::onControlCmd, this, _1));
-  sub_velocity_ = create_subscription<TwistStamped>(
-    "/localization/twist", 1, std::bind(&RawVehicleCommandConverterNode::onVelocity, this, _1));
+  sub_velocity_ = create_subscription<Odometry>(
+    "/localization/odometry", 1, std::bind(&RawVehicleCommandConverterNode::onVelocity, this, _1));
```
Comment (takayuki5168) [inline vehicle/raw_vehicle_cmd_converter/src/node.cpp:90]: Topic for odometory may be /localization/kinematic_state or something?

Comment (taikitanaka3) [reply vehicle/raw_vehicle_cmd_converter/src/node.cpp]:
```
/control/command/ackermann_control_command
/localization/kinematic_state
/parameter_events
/rosout
/vehicle/command/actuation_command
/vehicle/raw_vehicle_cmd_converter/debug/steer_pid
/vehicle/status/steering

```

Comment (takayuki5168) [reply vehicle/raw_vehicle_cmd_converter/src/node.cpp]: Ah you decided the topic name /control/command/ackermann_control_command!

**Comment by taikitanaka3** [comment]
@takayuki5168 
can take a look at?

File: vehicle/raw_vehicle_cmd_converter/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+## scenario_selector_node
+
+`raw_vehicle_command_converter` node is a node that convert desired acceleration and velocity to mechanical input by using feed forward + feed back control(optional).
+
+### Input topics
+
+| Name                         | Type                                                     | Description                                                                                                        |
+| ---------------------------- | -------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `~/input/control_cmd`        | autoware_auto_control_msgs::msg::AckermannControlCommand | target `velocity/acceleration/steering_angle/steering_angle_velocity` is necessary to calculate actuation command. |
+| `~/vehicle/status/steering"` | autoware_vehicle_msgs::SteeringReport                    | current status of steering used for steering feed back control                                                     |
```
Comment (tkimura4) [inline vehicle/raw_vehicle_cmd_converter/README.md:12]:
```suggestion
| `~/vehicle/status/steering"` | autoware_auto_vehicle_msgs::SteeringReport                    | current status of steering used for steering feed back control                                                     |
```

File: vehicle/raw_vehicle_cmd_converter/include/raw_vehicle_cmd_converter/node.hpp [hunk 4]
```diff
@@ -22,24 +22,25 @@
...
+#include <nav_msgs/msg/odometry.hpp>
 
 #include <memory>
 #include <string>
 #include <vector>
 
 namespace raw_vehicle_cmd_converter
 {
-using autoware_control_msgs::msg::ControlCommandStamped;
+using ControlCommandStamped = autoware_auto_control_msgs::msg::AckermannControlCommand;
```
Comment (takayuki5168) [inline vehicle/raw_vehicle_cmd_converter/include/raw_vehicle_cmd_converter/node.hpp:38]: It may cause misunderstanding? Don't you think so.

**Comment by takayuki5168** [comment]
LGTM except for the upper discussions.

File: vehicle/raw_vehicle_cmd_converter/include/raw_vehicle_cmd_converter/node.hpp (hunk 4 above)
Comment (taikitanaka3) [reply vehicle/raw_vehicle_cmd_converter/include/raw_vehicle_cmd_converter/node.hpp]: @takayuki5168 
change to using ackermann control command here a23e927

**Review by tkimura4 — APPROVED** [review]

**Pull request merged**